src/button-group/icon-button-item.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { forwardRef } from 'react';
+import clsx from 'clsx';
+
+import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
+
+import { ButtonProps } from '../button/interfaces.js';
+import { InternalButton } from '../button/internal.js';
+import LiveRegion from '../internal/components/live-region/index.js';
+import Tooltip from '../internal/components/tooltip/index.js';
+import { CancelableEventHandler, ClickDetail } from '../internal/events/index.js';
+import { ButtonGroupProps } from './interfaces.js';
+
+import styles from './styles.css.js';
+import testUtilStyles from './test-classes/styles.css.js';
+
+const IconButtonItem = forwardRef(
+  (
+    {
+      item,
+      showTooltip,
+      showFeedback,
+      onItemClick,
+    }: {
+      item: ButtonGroupProps.IconButton;
+      showTooltip: boolean;
+      showFeedback: boolean;
+      onItemClick?: CancelableEventHandler<ClickDetail>;
+    },
+    ref: React.Ref<ButtonProps.Ref>
+  ) => {
+    const containerRef = React.useRef<HTMLDivElement>(null);
+    const hasIcon = item.iconName || item.iconUrl || item.iconSvg;
+
+    if (!hasIcon) {
+      warnOnce('ButtonGroup', `Missing icon for item with id: ${item.id}`);
+    }
+
+    return (
+      <div ref={containerRef}>
+        <InternalButton
+          variant="icon"
+          loading={item.loading}
+          loadingText={item.loadingText}
+          disabled={item.disabled}
+          iconName={hasIcon ? item.iconName : 'close'}
```

Comment:
What's the motivation behind using the `close` icon as default variant? I'd like to better understand why the `iconName` property is optional and not required.

Reply:
Good question. It can be iconName, iconAlt or iconSvg. We can't make iconName requiried. 
@pan-kot waht do you think, should we display nothing if no icon defined?

Reply:
What about a `warnOnce` if none of them is provided?

Reply:
Yes, we already have warnOnce